operations/mimir/distributor.libsonnet

```diff
@@ -43,13 +43,17 @@
 
   local deployment = $.apps.v1.deployment,
 
+  // Don't allow all distributors to be unavailable.
+  local replica_count = 3,
+  local max_unavailable = std.min(1, if ($._config.autoscaling_distributor_enabled) then $._config.autoscaling_distributor_min_replicas - 1 else replica_count - 1),
+
   distributor_deployment: if !$._config.is_microservices_deployment_mode then null else
-    deployment.new('distributor', 3, [$.distributor_container]) +
+    deployment.new('distributor', replica_count, [$.distributor_container]) +
     $.newMimirSpreadTopology('distributor', $._config.distributor_topology_spread_max_skew) +
     $.mimirVolumeMounts +
     (if !std.isObject($._config.node_selector) then {} else deployment.mixin.spec.template.spec.withNodeSelectorMixin($._config.node_selector)) +
     deployment.mixin.spec.strategy.rollingUpdate.withMaxSurge(5) +
-    deployment.mixin.spec.strategy.rollingUpdate.withMaxUnavailable(1),
+    deployment.mixin.spec.strategy.rollingUpdate.withMaxUnavailable(max_unavailable),
```

Comment:
>Am I reading this right that if someone is running without distributor autoscaling and has set the number of replicas to 2, then this will set maxUnavailable: 2 still?

It's not possible to change the number of replicas if distributor autoscaling disabled - if distributor autoscaling is disabled, there will always be `replica_count` replicas, which is always 3.

However, @jhesketh pointed out in an internal PR for another similar that I can simplify this logic, so I've done that in 727269c

Reply:
> It's not possible to change the number of replicas if distributor autoscaling is disabled

The replicas can be overridden via jsonnet - `distributor_deployemnt+: { spec+: { replicas: 10 }}`

Reply:
Ah, good point :) I've fixed this in 04a9f6c.